components/rsptx/db/crud.py

```diff
@@ -1796,7 +1795,6 @@ async def fetch_one_user_topic_practice(
         & (UserTopicPractice.course_name == user.course_name)
         & (UserTopicPractice.chapter_label == last_page_chapter)
         & (UserTopicPractice.sub_chapter_label == last_page_subchapter)
-        & (UserTopicPractice.question_name == qname)
```

Comment:
I translated this from ajax.py and I thought it seemed wrong at the time, but Iman told me it was correct.

Reply:
I checked with Iman and understand how it works now; he agreed that it shouldn't take qname as a parameter.

Here's the logic. Each UserTopicPractice instance keeps track of one topic for one user. question_name is the last question practiced on that topic; when the topic is practiced it will go on to the next question associated with that topic.  (Or maybe it's encoding the next one that should be asked, and when the topic is practiced, it is set to be the next one to practice. Either way, it's the same for our purposes...)

When we *create* a UserTopicPractice, we need to set question_name to something. So the code sets it to be the name of the first question associated with the topic. This functino isn't called when we do that.

When we *delete* a UserTopicPractice for a topic, we use this function to look up the dbid of the row, then we delete that row_id.

This function doesn't seem to be referenced anywhere else in the repo. 

If we do end up moving the practice tool over at some point, it could make use of this function. But it will also want a version without the question_name in the query. Because it will be looking up the user, chapter, and sub-chapter, in order to find out what the curent question_name is.